CommandLine/emake/SOG.cpp

```diff
@@ -0,0 +1,69 @@
+#include "SOG.hpp"
+#include "compiler/event_reader/event_parser.h"
+
+#include <boost/filesystem.hpp>
+#include <boost/foreach.hpp>
+#include <fstream>
+#include <iostream>
+
+bool ReadSOG(const std::string &input_file, Game *game) {
+  event_parse_resourcefile();
+  map<string, evpair> event_filenames;
+  for (evpair e : event_sequence) {
+    event_filenames[event_get_function_name(e.first, e.second) + ".edl"] = e;
+    if (!e.second && event_is_instance(e.first, e.second)) {
+      for (int i = 1; i < 127; ++i) {
```

Comment:
What is 127 here? Please use constants when possible.

Reply:
The 127 is as arbitrary as it sounds. It needs to be bigger than the biggest vk_ constant, and big enough for alarms and user events, etc (which are eclipsed by the biggest vk_ constants). This is mostly a hack to work around the fact that the engine uses numbers instead of names for keys in keyboard events.

Reply:
if you don't have a name for it atleast leave a comment as to why you put it there so someone else reading this can follow your logic.

Reply:
That's fair; I will do so before I submit this, tonight.